sentenai/api/search.py

```diff
@@ -0,0 +1,390 @@
+            r = self.search._spans(cc)
+            c = r.get('cursor')
+            self.cursors.append(c)
+            rp = ResultPage(self.search, *[Result(self.search, **s) for s in r['spans']])
+            self.spans[cc] = rp
+            return next(self)
+
+    next = __next__
+
+
+    @property
+    def complete(self):
+        self.cursors[-1] is None
+
+
+    def df(self, *args, **kwargs):
+        dfs = []
+        for x in self[:]:
+            dfs.append(x.df(*args, **kwargs))
+        if len(dfs):
+            return pd.concat(dfs, keys=range(0,len(dfs)))
+        else:
+            return pd.DataFrame()
+
+
+    def all(self):
+        while self.cursors[-1]:
+            if self.cursors[-1] not in self.spans:
+                r = self.search._spans(self.cursors[-1])
+                c = r.get('cursor')
+                rp = ResultPage(self.search, *[Result(self.search, **s) for s in r['spans']])
+                self.spans[self.cursors[-1]] = rp
+                self.cursors.append(c)
+        return self
+
+
+    def _repr_html_(self):
+        data = []
+        for cursor in self.cursors:
+            data.extend(self.spans.get(cursor, []))
+        print([x.duration for x in data])
+        if data:
+            d = [ float(r.duration.total_seconds())
+                  for r in data
+                  if r.duration < timedelta.max]
+            maxd = max(d) if d else timedelta.max.total_seconds()
+
+        x = lambda r: int(round(r.duration.total_seconds() / maxd * 10)) * u"\u2588"
+        df = pd.DataFrame([
+            { 'Start': r.start,
+              'End': r.end,
+              'Duration': u"{}".format(r.duration if r.duration else 0),
+              'Viz': u"{}".format(x(r)),
+            } for r in data])
+        if df.empty:
+            return "_"
+        else:
+            return df[['Start', 'End', 'Duration', 'Viz']]._repr_html_()
+
+
+
+
+class Result(object):
+    def __init__(self, search, start, end, cursor):
+        self.search = search
+        self.start = cts(start) if start else None
+        self.end = cts(end) if end else None
+        self.events = 0
+        self.cursor = cursor
+        self.projection = None
+
+
+    @property
+    def duration(self):
+        if self.end and self.start:
+            return self.end - self.start
+        else:
+            return timedelta.max
+
+
+    def _repr_html_(self):
+        df = pd.DataFrame([
+            { 'Start': self.start,
+              'End': self.end,
+              'Duration': u"{}".format(self.duration if self.duration else 0),
+            }])
+        if df.empty:
+            return "_"
+        else:
+            return df[['Start', 'End', 'Duration']]._repr_html_()
+
+    def _count(self, max_retries=1):
+        if max_retries < 0:
+            raise SentenaiException("Max Retries Exceeded.")
+        streams = {}
+        url = '{host}/query/{cursor}/events'.format(host=self.search.client.host, cursor=self.cursor)
```

Comment:
does this `_count` method do anything? i don't see it called anywhere.

its implementation is also heavily copied from `_events` below but then it just returns a `0`. delete it?